go/chunks/chunk.go

```diff
@@ -6,78 +6,85 @@
 package chunks
 
 import (
-	"bytes"
-
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/hash"
+	"github.com/golang/snappy"
 )
 
-// Chunk is a unit of stored data in noms
+// Chunk is a unit of stored data in noms. Chunk data is compressed in storage
+// and on the wire.
 type Chunk struct {
-	r    hash.Hash
-	data []byte
+	h                        hash.Hash
+	uncompressed, compressed []byte
+	hashVerified             bool
 }
 
-var EmptyChunk = NewChunk([]byte{})
+var EmptyChunk = New([]byte{})
 
 func (c Chunk) Hash() hash.Hash {
-	return c.r
+	if !c.hashVerified {
```

Comment:
I feel like this has fundamentally changed the goroutine-safeness of Chunk. Is that a problem, or just something we need to document?

Reply:
Also, should `Chunk` be a pointer-receiver now?